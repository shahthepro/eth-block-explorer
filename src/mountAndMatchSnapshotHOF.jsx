export default function (it, ReactDOM, mount) {
  return function (elementName, element) {
    it(`renders <${elementName}> without crashing`, () => {
      const div = document.createElement('div');
      ReactDOM.render(element, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    
    it(`matches <${elementName}> snapshot`, () => {
      const mountedElement = mount(element);
      expect(mountedElement).toMatchSnapshot();
    });
  }
}