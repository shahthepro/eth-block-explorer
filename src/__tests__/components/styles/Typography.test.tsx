import * as React from 'react';
import * as ReactDOM from 'react-dom';
import {
  Subtitle, Title
} from './../../../components/styles/Typography';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import mountAndMatchSnapshotHOF from '../../../../mountAndMatchSnapshotHOF';

Enzyme.configure({ adapter: new Adapter() })
const { mount } = Enzyme;

const mountAndMatchSnapshot = mountAndMatchSnapshotHOF(it, ReactDOM, mount);

mountAndMatchSnapshot('Subtitle', <Subtitle />);
mountAndMatchSnapshot('Title', <Title />);