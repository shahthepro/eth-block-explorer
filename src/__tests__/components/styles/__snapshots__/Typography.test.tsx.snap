// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches <Subtitle> snapshot 1`] = `
<styled.h2>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "lastClassName": "QULuE",
          "rules": Array [
            "
        font-weight: 100;
        margin: 0;
        padding-bottom: 1rem;
      ",
          ],
        },
        "displayName": "styled.h2",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "h2",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <h2
      className="sc-bwzfXH QULuE"
    />
  </StyledComponent>
</styled.h2>
`;

exports[`matches <Title> snapshot 1`] = `
<styled.h1>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "gkfbEN",
          "rules": Array [
            "
        font-weight: 100;
        margin: 0;
        padding-bottom: 1rem;
        text-align: ",
            [Function],
            ";
      ",
          ],
        },
        "displayName": "styled.h1",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "h1",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <h1
      className="sc-bdVaJa gkfbEN"
    />
  </StyledComponent>
</styled.h1>
`;
